python/src/worker/tasks/background_expressed_genes.py

```diff
@@ -0,0 +1,128 @@
+from ..helpers.find_cells_by_set_id import find_cells_by_set_id
+from ..helpers.remove_regex import remove_regex
+from ..helpers.s3 import get_cell_sets
+from ..result import Result
+from . import Task
+
+
+class GetBackgroundExpressedGenes(Task):
+    def __init__(self, msg):
+        super().__init__(msg)
+        self.experiment_id = config.EXPERIMENT_ID
+        self.pagination = {}
+
+    def _format_result(self, result):
+        # Return a list of formatted results.
+        return Result({"genes": result["genes"]})
+
+    def _format_request(self):
+        # get cell sets from database
+        resp = get_cell_sets(self.experiment_id)
+
+        first_cell_set_name = self.task_def["cellSet"]
+        second_cell_set_name = self.task_def["compareWith"]
+        basis = self.task_def["basis"]
+
+        # Check if the comparsion is between all the cells or within a cluster
+        if not basis or ("all" in basis.lower()):
+            # In case that we are not going to filter by a cluster we remain the set
+            #  empty
+            filtered_set = set()
+        else:
+            # In the case that we filter by a cluster we keep the cells id in a set
+            filtered_set = set(find_cells_by_set_id(basis, resp))
+
+        # mark cells of first set
+        first_cell_set = find_cells_by_set_id(first_cell_set_name, resp)
+
+        # mark cells of second set
+        # check if the second set is composed by the "All other cells"
+        if (
+            second_cell_set_name == "background"
+            or "all" in second_cell_set_name.lower()
+        ):
+            # Retrieve all the cells (not necessary at the same hierachy level)
+            complete_cell_set = set(find_all_cell_ids_in_cell_sets(resp))
+            # Filter with those that are not in the first cell set
+            second_cell_set = [
+                item for item in complete_cell_set if item not in first_cell_set
+            ]
+        else:
+            # In the case that we compare with specific cell set, we just look for
+            #  the cell directly
+            second_cell_set = self.get_cells_in_set(
+                second_cell_set_name, resp, first_cell_set_name
+            )
+            # Check any possible intersect cells
+            inter_cell_set = set(first_cell_set).intersection(set(second_cell_set))
+            first_cell_set = [
+                item for item in first_cell_set if item not in inter_cell_set
+            ]
+            second_cell_set = [
+                item for item in second_cell_set if item not in inter_cell_set
+            ]
+
+        # Keep only the cell_set that are on the specify basis (in the case that we
+        # are not in the "All" analysis)
+        if len(filtered_set) > 0:
+            second_cell_set = [it for it in second_cell_set if it in filtered_set]
+            first_cell_set = [it for it in first_cell_set if it in filtered_set]
```

Comment:
this look like set intersections, which are [much faster](https://stackoverflow.com/a/17945009). Also reorder for OCD reasons

```suggestion
            first_cell_set = list(first_cell_set.intersection(filtered_set))
            second_cell_set = list(second_cell_set.intersection(filtered_set))
```

Reply:
changed to use set operation